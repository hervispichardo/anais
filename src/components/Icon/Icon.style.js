import styled from 'styled-components';


const Ripple = styled.div`
  border: 0px !important;
  ${({ ripple }) => ripple && `
    border-radius: 50%;
    cursor: pointer;
    padding: ${ ({ padding }) => padding || '5px'};
    display: inline-block;
    background-position: center;
    transition: background 0.8s;
    &:before {
      border: none !important;
    }
    &:hover {
      background: transparent radial-gradient(circle, transparent 1%, rgba(0, 0, 0, 0.1) 1%) center/15000%;
    }
    &:active {
      background-color: rgba(0, 0, 0, 0.2);
      background-size: 100%;
      transition: background 0s;
      &:after {
        background: #000;
        content: '';
        height: 20px;
        width: 20px;
      }
    }
  `}

`;


export default Ripple

export {
  Ripple
 }