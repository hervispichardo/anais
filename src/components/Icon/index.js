import React from 'react';
import PropTypes from 'prop-types'
import Icons from './icons'
import { Ripple } from './style'


const Icon = ({ ripple, style, color, size, type, badge, badgeColor }) => {
  const styles = {
    svg: {
      display: 'inline-block',
      verticalAlign: 'middle',
    },
    path: {
      fill: color,
    },
    badge: {
      fill: badgeColor,
    },
    circle: {
      fill: '#fff',
    },
  };

  const badgeNumber = badge > 8 ? 'plus' : badge;

  return (
    <Ripple ripple={ripple}>
      <svg
        style={{ ...styles.svg, ...style }}
        width={`${size}em`}
        height={`${size}em`}
        viewBox="0 0 1024 1024"
      >
        <g>
          {
            Icons[type] && Icons[type].path
              ? <path style={styles.path} d={ Icons[type].path} />
              :  Icons[type] && Icons[type].markup
          }
          {
            badge &&
            <g>
              <circle style={styles.circle} cx="310" cy="290" r="270" />
              <path style={styles.badge} d={Icons.badge[badgeNumber]} />
            </g>
          }
        </g>
      </svg>
    </Ripple>
  );
};

Icon.propTypes = {
  size: PropTypes.number,
  type: PropTypes.string,
  color: PropTypes.string.isRequired,
  ripple: PropTypes.boolean,
  badge: PropTypes.number,
  badgeColor: PropTypes.string,
};

Icon.defaultProps = {
  size: 1.8,
  color: '#fff',
  badge: 0,
  badgeColor: '#08798c',
  ripple: true,
};

export default Icon
export { Icons }
