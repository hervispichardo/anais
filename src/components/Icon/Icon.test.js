import React from 'react'
import renderer from 'react-test-renderer'
import { shallow, mount } from 'enzyme'
import Icon, { Icons } from './Icon'

it('render correctly Icon component', () => {
  const iconComponent = renderer.create(
    <Icon
      color={'#000'}
      type={Icons.calendar}
    />).toJSON();
  expect(iconComponent).toMatchSnapshot();
});

