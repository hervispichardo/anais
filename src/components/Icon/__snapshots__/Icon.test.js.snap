// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`render correctly Icon component 1`] = `
<div
  className="ripple"
>
  <svg
    height="1.8em"
    style={
      Object {
        "display": "inline-block",
        "verticalAlign": "middle",
      }
    }
    viewBox="0 0 1024 1024"
    width="1.8em"
  >
    <g>
      <path
        d="M285,50 C301.568542,50 315,63.4314575 315,80 L315,200 C315,216.568542 301.568542,230 285,230 C268.431458,230 255,216.568542 255,200 L255,80 C255,63.4314575 268.431458,50 285,50 Z M725,50 C741.568542,50 755,63.4314575 755,80 L755,200 C755,216.568542 741.568542,230 725,230 C708.431458,230 695,216.568542 695,200 L695,80 C695,63.4314575 708.431458,50 725,50 Z M285,531 C251.862915,531 225,504.137085 225,471 C225,437.862915 251.862915,411 285,411 C318.137085,411 345,437.862915 345,471 C345,504.137085 318.137085,531 285,531 Z M505,531 C471.862915,531 445,504.137085 445,471 C445,437.862915 471.862915,411 505,411 C538.137085,411 565,437.862915 565,471 C565,504.137085 538.137085,531 505,531 Z M725,531 C691.862915,531 665,504.137085 665,471 C665,437.862915 691.862915,411 725,411 C758.137085,411 785,437.862915 785,471 C785,504.137085 758.137085,531 725,531 Z M285,741 C251.862915,741 225,714.137085 225,681 C225,647.862915 251.862915,621 285,621 C318.137085,621 345,647.862915 345,681 C345,714.137085 318.137085,741 285,741 Z M505,741 C471.862915,741 445,714.137085 445,681 C445,647.862915 471.862915,621 505,621 C538.137085,621 565,647.862915 565,681 C565,714.137085 538.137085,741 505,741 Z M725,741 C691.862915,741 665,714.137085 665,681 C665,647.862915 691.862915,621 725,621 C758.137085,621 785,647.862915 785,681 C785,714.137085 758.137085,741 725,741 Z M157.873134,158 C142.479227,158 130,170.529345 130,185.985075 L130,880.014925 C130,895.470655 142.479227,908 157.873134,908 L849.126866,908 C864.520773,908 877,895.470655 877,880.014925 L877,185.985075 C877,170.529345 864.520773,158 849.126866,158 L157.873134,158 Z M157.571429,102 L848.428571,102 C894.583797,102 932,139.590638 932,185.961039 L932,880.038961 C932,926.409362 894.583797,964 848.428571,964 L157.571429,964 C111.416203,964 74,926.409362 74,880.038961 L74,185.961039 C74,139.590638 111.416203,102 157.571429,102 Z"
        style={
          Object {
            "fill": "#000",
          }
        }
      />
      
    </g>
  </svg>
</div>
`;
